drivers/SmartThings/zwave-thermostat/src/ct100-thermostat/init.lua

```diff
@@ -13,212 +13,73 @@
 -- limitations under the License.
 
 local capabilities = require "st.capabilities"
-local utils = require "st.utils"
---- @type st.zwave.CommandClass
 local cc = require "st.zwave.CommandClass"
 --- @type st.zwave.CommandClass.SensorMultilevel
-local SensorMultilevel = (require "st.zwave.CommandClass.SensorMultilevel")({ version = 5 })
---- @type st.zwave.CommandClass.ThermostatFanMode
-local ThermostatFanMode = (require "st.zwave.CommandClass.ThermostatFanMode")({ version = 3 })
---- @type st.zwave.CommandClass.ThermostatFanState
-local ThermostatFanState = (require "st.zwave.CommandClass.ThermostatFanState")({ version = 2 })
+local SensorMultilevel = (require "st.zwave.CommandClass.SensorMultilevel")({ version = 2 })
 --- @type st.zwave.CommandClass.ThermostatMode
 local ThermostatMode = (require "st.zwave.CommandClass.ThermostatMode")({ version = 2 })
 --- @type st.zwave.CommandClass.ThermostatOperatingState
 local ThermostatOperatingState = (require "st.zwave.CommandClass.ThermostatOperatingState")({ version = 1 })
 --- @type st.zwave.CommandClass.ThermostatSetpoint
 local ThermostatSetpoint = (require "st.zwave.CommandClass.ThermostatSetpoint")({ version = 1 })
+--- @type st.zwave.CommandClass.ThermostatFanMode
+local ThermostatFanMode = (require "st.zwave.CommandClass.ThermostatFanMode")({ version = 1 })
+--- @type st.zwave.CommandClass.Battery
+local Battery = (require "st.zwave.CommandClass.Battery")({ version = 1 })
+local constants = require "st.zwave.constants"
+local utils = require "st.utils"
 
 local CT100_THERMOSTAT_FINGERPRINTS = {
   { manufacturerId = 0x0098, productType = 0x6401, productId = 0x0107 }, -- 2Gig CT100 Programmable Thermostat
   { manufacturerId = 0x0098, productType = 0x6501, productId = 0x000C }, -- Iris Thermostat
 }
 
--- Constants
-local TEMPERATURE_SCALE = "temperature_scale"
-local PRECISION = "precision"
-local CURRENT_HEATING_SETPOINT = "current_heating_setpoint"
-local CURRENT_COOLING_SETPOINT = "currnet_cooling_setpoint"
-local MODE = "mode"
-local TEMPERATURE = "temperature"
-local HEATING_SETPOINT_IS_LIMITED = "heating_setpoint_is_limited"
-local COOLING_SETPOINT_IS_LIMITED = "cooling_setpoint_is_limited"
-
-local MIN_HEATING_SETPOINT = 35.0
-local MAX_HEATING_SETPOINT = 92.0
-local MIN_COOLING_SETPOINT = 38.0
-local MAX_COOLING_SETPOINT = 95.0
+-- This old device uses separate endpoints to get values of temp and humidity
+-- DTH actually uses the old mutliInstance encap, but multichannel should be back-compat
+local TEMPERATURE_ENDPOINT = 1
+local HUMIDITY_ENDPOINT = 2
 
-local function can_handle_ct100_thermostat(opts, driver, device, cmd, ...)
-  for _, fingerprint in ipairs(CT100_THERMOSTAT_FINGERPRINTS) do
-    if device:id_match( fingerprint.manufacturerId, fingerprint.productType, fingerprint.productId) then
-      return true
-    end
+--TODO: Update this once we've decided how to handle setpoint commands
+local function convert_to_device_temp(command_temp, device_scale)
```

Comment:
I know this is a workaround, but would it be better to set the cutoff to 35 in this case since that is the highest celcius value we would expect for this subdriver based on the profile?

Reply:
sure. that makes sense

Reply:
here's hoping nobody ever sets their thermostat to 35 C or F, or 40 C or F